Emails: Prevent placeholder from showing on Email page when site domains are fetching

stephanethomas opened the pull request:
This pull request updates the `Email` page to only show a placeholder when G Suite users and site domains have not loaded yet. This solves an issue where the placeholder could be displayed twice or more as soon as site domains are fetched more than once:

![screenshot](https://user-images.githubusercontent.com/594356/110462142-979af580-80d0-11eb-8d0d-638e8338cbe7.png)


This pull request also makes the following improvements:

* Remove unnecessary indentation in stylesheets
* Simplify imports of Titan functions
* Fix invalid translation comment in `Add New Users` page

#### Testing instructions

1. Run `git checkout update/email-page` and start your server, or open a [live branch](https://calypso.live/?branch=update/email-page)
2. Log into a WordPress.com account with an email account
3. Open the [`Domains` page](http://calypso.localhost:3000/domains)
4. Click on a domain with mailboxes
5. Click the `Manage your email accounts` option
6. Assert that the placeholder is displayed for a short time
7. Click the `Back` button in the section header
8. Click the `Manage your email accounts` option again
9. Assert that no placeholder is displayed this time

stephanethomas commented on client/my-sites/email/email-management/gsuite-user-item/index.jsx:
```diff
@@ -70,7 +70,7 @@ function GSuiteUserItem( props ) {
 		return (
 			<Fragment>
-				<Button compact={ true } onClick={ onFixClickHandler }>
+				<Button compact onClick={ onFixClickHandler }>
```
There is no need to pass `true` so I've simplified that line.

stephanethomas commented on client/my-sites/email/email-management/gsuite-users-card/style.scss:
```diff
@@ -37,54 +37,54 @@
 	margin-bottom: 0;
+}
 
-	.gsuite-users-card__foldable-card-header-icon {
```
I have made no changes to that file except removing unnecessary indentation - as recommended [here](https://github.com/Automattic/wp-calypso/blob/trunk/docs/components.md#syntax).

stephanethomas commented on client/my-sites/email/email-management/index.jsx:
```diff
@@ -130,15 +126,9 @@ class EmailManagement extends React.Component {
-		if ( ! hasGSuiteUsersLoaded || ! hasSiteDomainsLoaded || isFetchingSiteDomains ) {
+		const { domains, hasGSuiteUsersLoaded, hasSiteDomainsLoaded, selectedDomainName } = this.props;
+
+		if ( ! hasGSuiteUsersLoaded || ! hasSiteDomainsLoaded ) {
```
I've updated that code to avoid displaying the placeholder when site domains are fetching. This solves an issue when we reload the `Email` page where the placeholder would be displayed twice.

stephanethomas commented on client/my-sites/email/email-management/titan-management-nav/style.scss:
```diff
@@ -7,56 +7,64 @@
+	span {
+		flex-grow: 1;
+	}
+}
```
I have made no changes to that file except removing unnecessary indentation, and adding rules for `titan-management-nav__mailboxes-card`.

hambai reviewed (approved):
Okay tested this and it looks good

hambai reviewed (changes requested):
Actually I just reviewed #50515 and Dale is opening the "finish setup" link in new window. I believe for consistency we should do the same here.

Also his PR works with the `titan/iframe-control-panel` flag - should this PR also work with the iframe version?

hambai commented:
Okay okay - I got a bit confused with all the PRs. I guess the original task from this PR is implemented in #50527 but there are some other small improvements here like how we import stuff from `calypso/lib/titan` and some SCSS updates that we still want to land but we'll need to get #50527 and #50515 landed first and rebase this PR after

stephanethomas committed:
Emails: Remove unnecessary indentation in stylesheets

stephanethomas committed:
Emails: Prevent placeholder from showing when site domains are fetching

We should only show it when G Suite users and site domains has not loaded yet. This solves an issue where the placeholder would be displayed twice when the Email page is reloaded manually.

stephanethomas committed:
Emails: Simplify imports of Titan functions

stephanethomas committed:
Emails: Fix invalid translation comment in Add New Users page

stephanethomas commented:
Now that https://github.com/Automattic/wp-calypso/pull/50527 has been merged, I'm going to clean up that pull request and repurpose it so it only caters minor changes and refactorings.

olaseni reviewed (approved):
LGTM! Works as described.
